### ISSUE TITLE
Add setting for configuring database & cache locations

### ISSUE THREAD

**Issue by Frooxius** [Yellow-Dog-Man/Resonite-Issues#1985]
### Is your feature request related to a problem? Please describe.

Currently the location of cache & database is fixed and can only be configured via startup parameters. It will be useful to be able to configure the locations directly in-game.

### Describe the solution you'd like

Add settings for configuring:
- Cache location
- Persistent Data & database location

It will also be useful to add option that will migrate any data from old location to the new one.

### Describe alternatives you've considered

Keep configuration through command-line arguments or config file, but those are less user friendly.

**Comment by shadowpanther** [comment]
The question is, how would you store the location if the data/database folder location is stored in the database?

Also, settings seem to differ for local user and logged-in user, so if you log out the different set of folders might be used.

**Comment by epicEaston197** [comment]
> I think it might also important to make sure this is setting is per device. If I play Resonite on multiple computers I might want to change to a different directory on each.

Yeah that would make a lot of sense especially because you don't want resonite writing data to a drive that doesn't exist

**Comment by Frooxius** [comment]
@shadowpanther Likely a small config file in the "normal" location, which indicates where it should be stored last.

@JackTheFoxOtter Yes, this will implicitly be per device and non-overridable.
